source/infrastructure/pipeline/e2e-tests/utils/schedule-test-utils.ts

```diff
@@ -0,0 +1,86 @@
+/*****************************************************************************
+ *  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.   *
+ *                                                                            *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may   *
+ *  not use this file except in compliance with the License. A copy of the    *
+ *  License is located at                                                     *
+ *                                                                            *
+ *      http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                            *
+ *  or in the 'license' file accompanying this file. This file is distributed *
+ *  on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,        *
+ *  express or implied. See the License for the specific language governing   *
+ *  permissions and limitations under the License.                            *
+ *****************************************************************************/
+import * as dynamodb from "@aws-sdk/client-dynamodb"
+import {hubStackParams} from "./hub-stack-utils";
+
+const configTableName = hubStackParams.configTableArn.substring(hubStackParams.configTableArn.lastIndexOf("/") + 1)
+export interface Period {
+  name: string,
+  description: string
+  begintime: string,
+  endtime: string
+}
+
+export interface Schedule {
+  name: string
+  description: string
+  periods: Period[]
+}
+export async function createSchedule(client: dynamodb.DynamoDBClient, schedule: Schedule) {
+  await client.send(
+    new dynamodb.BatchWriteItemCommand({
+      RequestItems: {
+        [configTableName]: scheduleToBatchedPutRequests(schedule)
+      }
+    })
+  )
+}
```

Comment:
Can we have some unit tests for these functions?

Reply:
How would I go about doing that? These are utilities for simplifying the interface with dynamodb so any tests on them would essentially be another set of full e2e tests.

Since they are purely testing utilities that are depended on by all the other e2e tests that will use dynamo I don't see much benefit to testing them separately as any failures with them would cause cascade failures in all the other e2e tests, and a few more failing tests would probably just add to the noise?

Reply:
I'm not saying to add more e2e tests. I'm saying to add unit tests that run as part of the unit test/build stage. You're assuming that a bug in one of your testing utilities would lead to failing tests. What if you introduce a bug that gives bad information to your tests in such a way where they always pass?

The pipeline is also purely for testing, and it should also have unit tests, and probably integration tests.